src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -155,12 +177,18 @@ public void startMonitor() {
         }
         subscribeTaskFuture = executorService.submit(() -> {
             try {
-                subscribeToShadowTopics();
-                publishToGetCISShadowTopic();
+                subscribeToCISShadowTopicsWithRetry();
             } catch (InterruptedException e) {
                 LOGGER.atDebug().cause(e).log("Interrupted while subscribing to CIS shadow topics");
                 Thread.currentThread().interrupt();
+                return;
+            } catch (Exception e) {
+                // retries exhausted
+                // TODO propagate error to service level
```

Comment:
planning on addressing this in a separate PR, I'd imagine we want to mark the service as errored if we aren't able to react to CIS changes